ERCS/erc-7682.md

```diff
@@ -99,15 +99,56 @@ type WalletSendCallsRequest = {
   capabilities?: {
     auxiliaryFunds?: {
       assetsRequiredForExecution?: {
-        [asset: `0x${string}`]: {
-          amount: `0x${string}`; // Amount required, as a hex string representing the integer value in the asset's smallest unit
+        erc20?: {
+          [asset: `0x${string}`]: {
+            amount: `0x${string}`; // Amount required, as a hex string representing the integer value in the asset's smallest unit
+          };
+        };
+        erc721?: {
+          [asset: `0x${string}`]: {
+            tokenId: `0x${string}`; // Token ID as a hex string
+          };
+        };
+        erc1155?: {
```

Comment:
Just confirming that we've gone with this standard as key approach to make it explicit that `erc20` doesn't support `tokenId`?

The alternative could be an array of shared schema such as:

```
assetsRequiredForExecution?: {
    asset: `0x${string}`;
    amount: `0x${string}`;
    tokenId?: `0x${string}`;
}[]
```

This has the advantage of simplicity, and meaning we wouldn't necessarily have to define specific standards in this specification, meaning it is decoupled and the wallet could support future standards?

One downside is the risk of duplicate assets.

Reply:
Yeah this was closer to the format I'd originally proposed. I agree its probably cleaner than building in certain specification for the 3 primary token standards standards we anticipate. The main issue I see with it is that we aren't fully avoiding the possible need for extensions in the future since the properties we're assuming here (`amount` and `tokenId`) are the superset of the required properties for these specific standards. Any new standards with additionally required properties would need to propose an extension

Reply:
Reverted to your suggested format: https://github.com/ethereum/ERCs/pull/1126/commits/55e7241c3fa69911ccd91b4d8642f542869dd9e9